transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java

```diff
@@ -168,53 +168,69 @@ protected void run() {
     @Override
     public boolean handle(int fd, int res, long flags, int op, int pollMask) {
         IOUringSubmissionQueue submissionQueue = ringBuffer.getIoUringSubmissionQueue();
+        final AbstractIOUringChannel channel;
```

Comment:
Could do as a follow-on but this would probably be cleaner to cover non-channel-specific cases first, then lookup the channel once and pass it to appropriate handler if non-null (rather than looking up separately in each of the handlers)

Reply:
let me do a followup